prog.go

```diff
@@ -585,7 +585,9 @@ func (p *Program) Benchmark(in []byte, repeat int, reset func()) (uint32, time.D
 
 var haveProgTestRun = internal.FeatureTest("BPF_PROG_TEST_RUN", "4.12", func() error {
 	prog, err := NewProgram(&ProgramSpec{
-		Type: SocketFilter,
+		// The first PROG_TEST_RUN patches shipped in 4.12 only supported testing
+		// SKB and XDP programs.
+		Type: SchedACT,
```

Comment:
I changed this to SocketFilter since that can be run without privileges on newer kernels. What if we keep SocketFilter add a special case for ENOTSUPP here as well?

Reply:
ENOTSUPP in this case indicating that we have prog run support, since we otherwise would get EINVAL.

Reply:
Yep, this works too!